File: Wdtc/src/main/java/org/wdt/wdtc/ui/window/user/NewUserWindows.java
package org.wdt.wdtc.ui.window.user;

import com.jfoenix.controls.JFXButton;
import javafx.scene.Scene;
import javafx.scene.image.Image;
import javafx.scene.layout.AnchorPane;
import javafx.scene.layout.Pane;
import javafx.stage.Modality;
import javafx.stage.Stage;
import org.wdt.wdtc.core.auth.accounts.Accounts;
import org.wdt.wdtc.core.manger.URLManger;
import org.wdt.wdtc.core.utils.URLUtils;
import org.wdt.wdtc.ui.window.Consoler;

public class NewUserWindows {
  private final Stage MainStage;
  private Accounts.AccountsType type = null;
  private String title;

  public NewUserWindows(Stage mainStage) {
    MainStage = mainStage;
  }

  public void setTitle(String title) {
    this.title = title;
  }

  public void setType(Accounts.AccountsType type) {
    this.type = type;
  }

  public void show() {
    Stage UserStage = new Stage();
    UserStage.setWidth(725);
    UserStage.setHeight(430);
    AnchorPane ParentPane = new AnchorPane();
    UserStage.getIcons().add(new Image("/assets/icon/ico.jpg"));
    Pane SonPane = new Pane();
    if (type == null) {
      UserListPane.setUserList(SonPane);
    } else {
      switch (type) {
        case Yggdrasil -> LittleskinWindow.setLittleskinWin(SonPane);
        case Offline -> OfflineUserWindow.setUserWin(SonPane);
      }
    }
    UserStage.initOwner(MainStage);
    UserStage.setTitle(title);
    SonPane.setPrefSize(600, 400);
    AnchorPane ButtonLine = new AnchorPane();
    ButtonLine.setPrefWidth(125);

    JFXButton UserListButton = new JFXButton("账户列表");
    UserListButton.setOnAction(event -> UserListPane.setUserList(SonPane));
    UserListButton.setPrefSize(125, 30);

    JFXButton OfflineButton = new JFXButton("离线账户");
    AnchorPane.setTopAnchor(OfflineButton, 30.0);
    OfflineButton.setPrefSize(125, 30);
    OfflineButton.setOnAction(event -> OfflineUserWindow.setUserWin(SonPane));
    JFXButton LittleskinCom = new JFXButton("Littleskin官网");
    LittleskinCom.setPrefSize(125, 30);
    AnchorPane.setBottomAnchor(LittleskinCom, 30.0);
    JFXButton buygame = new JFXButton("购买正版");
    buygame.setPrefSize(125, 30);
    AnchorPane.setBottomAnchor(buygame, 0.0);
    JFXButton YggdrasilButton = new JFXButton("Yggdrasil账户");
    YggdrasilButton.setPrefSize(125, 30);
    YggdrasilButton.setOnAction(event -> LittleskinWindow.setLittleskinWin(SonPane));
    AnchorPane.setTopAnchor(YggdrasilButton, 60.0);
    Consoler.setTopLowerRight(SonPane);
    AnchorPane.setBottomAnchor(SonPane, 0.0);
    AnchorPane.setLeftAnchor(SonPane, 125.0);
    ButtonLine.getChildren().addAll(UserListButton, OfflineButton, YggdrasilButton, LittleskinCom, buygame);
    ButtonLine.getStyleClass().add("BlackBorder");
    Consoler.setTopLowerLeft(ButtonLine);
    AnchorPane.setTopAnchor(ButtonLine, 0.0);
    ParentPane.setBackground(Consoler.getBackground());
    ParentPane.getChildren().addAll(ButtonLine, SonPane);
    UserStage.setScene(new Scene(ParentPane));
    Consoler.setStylesheets(ParentPane);
    UserStage.initModality(Modality.APPLICATION_MODAL);
    UserStage.setResizable(false);
    UserStage.show();
    LittleskinCom.setOnAction(event -> URLUtils.openSomething(URLManger.getLittleskinUrl()));
    buygame.setOnAction(event -> URLUtils.openSomething("https://www.minecraft.net/zh-hans"));
  }


}

File: Wdtc/src/main/kotlin/org/wdt/wdtc/ui/window/user/NewUserWindows.kt
package org.wdt.wdtc.ui.window.user

import com.jfoenix.controls.JFXButton
import javafx.event.EventHandler
import javafx.scene.Scene
import javafx.scene.image.Image
import javafx.scene.layout.AnchorPane
import javafx.scene.layout.Pane
import javafx.stage.Modality
import javafx.stage.Stage
import org.wdt.wdtc.core.auth.accounts.Accounts.AccountsType
import org.wdt.wdtc.core.manger.littleskinUrl
import org.wdt.wdtc.core.utils.openSomething
import org.wdt.wdtc.ui.window.*

class NewUserWindows(private val mainStage: Stage) {
  var type: AccountsType? = null
  var title: String? = null

  fun show() {
    val userStage = Stage().apply {
      width = 725.0
      height = 430.0
      icons.add(Image("/assets/icon/ico.jpg"))
      initOwner(mainStage)
      title = this@NewUserWindows.title
      initModality(Modality.APPLICATION_MODAL)
      isResizable = false
    }
    val sonPane = Pane().apply {
      setTopLowerRight()
      setBottomAnchor(0.0)
      setLeftAnchor(125.0)
      setPrefSize(600.0, 400.0)
    }
    sonPane.let {
      if (type == null) {
        UserListPane.setUserList(it)
      } else {
        when (type) {
          AccountsType.YGGDRASIL -> LittleskinWindow.setLittleskinWin(it)
          AccountsType.OFFLINE -> OfflineUserWindow.setUserWin(it)
          else -> TODO()
        }
      }
    }
    val userListButton = JFXButton("账户列表").apply {
      onAction = EventHandler { UserListPane.setUserList(sonPane) }
      setPrefSize(125.0, 30.0)
    }
    val offlineButton = JFXButton("离线账户").apply {
      setTopAnchor(30.0)
      setPrefSize(125.0, 30.0)
      onAction = EventHandler { OfflineUserWindow.setUserWin(sonPane) }
    }
    val littleskinCom = JFXButton("Littleskin官网").apply {
      setPrefSize(125.0, 30.0)
      setBottomAnchor(30.0)
      onAction = EventHandler { openSomething(littleskinUrl) }
    }
    val buygame = JFXButton("购买正版").apply {
      setPrefSize(125.0, 30.0)
      setBottomAnchor(0.0)
      onAction = EventHandler { openSomething("https://www.minecraft.net/zh-hans") }
    }
    val yggdrasilButton = JFXButton("Yggdrasil账户").apply {
      setPrefSize(125.0, 30.0)
      onAction = EventHandler { LittleskinWindow.setLittleskinWin(sonPane) }
      setTopAnchor(60.0)
    }
    val buttonLine = AnchorPane().apply {
      prefWidth = 125.0
      children.addAll(userListButton, offlineButton, yggdrasilButton, littleskinCom, buygame)
      styleClass.add("BlackBorder")
      setTopLowerLeft()
      setTopAnchor(0.0)
    }
    AnchorPane().apply {
      background = wdtcBackground
      children.addAll(buttonLine, sonPane)
      setStylesheets()
    }.let {
      userStage.run {
        scene = Scene(it)
        show()
      }
    }
  }
}
